docs/getting-started/integrate-auth/dotnet.mdx

```diff
@@ -0,0 +1,277 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+// create a new Ory Client with the BasePath set to the Ory Tunnel enpoint
+var oryBasePath = builder.Configuration.GetValue<string>("ORY_BASEPATH") ?? "http://localhost:4000";
+var ory = new FrontendApi(new Configuration
+{
+	BasePath = oryBasePath
+});
+
+// add session middleware
+app.Use(async (ctx, next) =>
+{
+	async Task Login()
+	{
+		// this will redirect the user to the managed Ory Login UI
+		var flow = await ory.CreateBrowserLoginFlowAsync() ?? throw new InvalidOperationException("Could not create browser login flow");
+		ctx.Response.Redirect(flow.RequestUrl);
+	}
+
+	try
+	{
+		// check if we have a session
+		var session = await ory.ToSessionAsync(cookie: ctx.Request.Headers.Cookie, cancellationToken: ctx.RequestAborted);
+		if (session?.Active is not true)
+		{
+			await Login();
+			return;
+		}
+
+		// add session to HttpContext
+		ctx.Items["req.session"] = session;
+	}
+	catch (ApiException)
+	{
+		await Login();
+		return;
+	}
+
+	await next(ctx);
+});
+
+app.MapGet("/", () => "Hello World!");
+
+app.Run();
+```
+
+## The protected page
+
+As the final step, create an `Index` page that's presented to signed-in users. This page diplays the user's session data.
+
+### Create the page
+
+Create the `Index` RazorPage that renders the page with the session data:
+
+```shell-session
+dotnet new page --name Index --no-pagemodel --output Pages
+```
+
+Add this code to the `Pages/Index.cshtml` file to present the data to the user:
+
+```cshtml title="Pages/Index.cshtml" {3-4,7-17}
+@page
+@{
+  // try to retrieve session from HttpContext
+  var session = HttpContext.Items.TryGetValue("req.session", out var item) && item is Ory.Client.Model.ClientSession s ? s.ToJson() : "<NO SESSION>";
+}
+
+<html lang="en">
+  <head>
+    <title>Ory Network secured ASP.NET Core web app</title>
+  </head>
+  <body>
+    <h1>Dashboard</h1>
+    <hr />
+    <h2>Your Session Data:</h2>
+    <pre><code>@session</code></pre>
+  </body>
+</html>
+```
+
+### App configuration
+
+To configure the app to serve RazorPages, make the following changes in `Program.cs`:
+
+```csharp title="Program.cs" showLineNumbers {3-4,21-22}
+var builder = WebApplication.CreateBuilder(args);
+
+// add support for RazorPages
+builder.Services.AddRazorPages();
+
+var app = builder.Build();
+
+// create a new Ory Client with the BasePath set to the Ory Tunnel enpoint
+var oryBasePath = builder.Configuration.GetValue<string>("ORY_BASEPATH") ?? "http://localhost:4000";
+var ory = new FrontendApi(new Configuration
+{
+	BasePath = oryBasePath
+});
+
+// add session middleware
+app.Use(async (ctx, next) =>
+{
+	// ... session middleware logic ...
+});
+
+// configure pipeline to use RazorPages
+app.MapRazorPages();
+
+app.Run();
+```
+
+## Test your application
+
+With all of the pieces in place, it's time to test your application. Follow these steps:
+
+1. Start your ASP.NET app:
```

Comment:
I followed the tutorial exactly to this point and get this:

```
$ dotnet run
Building...
/Users/arneluenser/dev/dotnet-tutorial/Program.cs(9,15): error CS0246: The type or namespace name 'FrontendApi' could not be found (are you missing a using directive or an assembly reference?) [/Users/arneluenser/dev/dotnet-tutorial/dotnet-tutorial.csproj]
/Users/arneluenser/dev/dotnet-tutorial/Program.cs(9,31): error CS0246: The type or namespace name 'Configuration' could not be found (are you missing a using directive or an assembly reference?) [/Users/arneluenser/dev/dotnet-tutorial/dotnet-tutorial.csproj]
/Users/arneluenser/dev/dotnet-tutorial/Program.cs(35,9): error CS0246: The type or namespace name 'ApiException' could not be found (are you missing a using directive or an assembly reference?) [/Users/arneluenser/dev/dotnet-tutorial/dotnet-tutorial.csproj]

The build failed. Fix the build errors and run again.
```

Reply:
I have left out the namespace imports in the code boxes on purpose to remove noise. Maybe they should be added to make the tutorial truly self contained?

It's only two lines of Code, so I will add the imports.